Lib/concurrent/futures/_base.py

```diff
@@ -606,6 +622,12 @@ def result_iterator():
                 # reverse to keep finishing order
                 fs.reverse()
                 while fs:
+                    if (
+                        buffersize
+                        and (executor := executor_weakref())
+                        and (args := next(args_iter, None))
```

Comment:
`args` may be empty, so you need to check for `args is not None`

Reply:
Are you refering to the case where one call `executor.map(func)` without any input iterable?

Reply:
you are right! But in such a case we will not enter the `while fs:` (`fs` being empty in that case), right?

Reply:
> But in such a case we don't enter the while fs.

Not necessarily. What I meant is that you call `executor.map` with an input iterable that yields `args = ()` everytime.

Note that it also doesn't hurt to check `is not None` because it's probably slightly faster since otherwise you need to call `__bool__` on the args being yielded.

Reply:
> You may have missed the ziping in your reasoning?

I did :) Sorry, my bad! 

> do you feel that (args := next(args_iter, None)) is not None is more natural?

I feel it would at least help avoiding questions like mine! (and it would still be probably slightly better performance wise but this claim is just my gut feeling).

Reply:
Yes I see... I have renamed `args_iter` into a more self-explanatory `zipped_iterable`, would it be enough to avoid the confusion in your opinion? 

(Because I am scared that the addition of `is not None` may misslead some of our fellow pythonistas wondering "wait, why is this not None check necessary here, what am I missing here 🤔?")